pkg/dockerregistry/server/manifestservice.go

```diff
@@ -42,6 +42,10 @@ func (m *manifestService) Exists(ctx context.Context, dgst digest.Digest) (bool,
 
 	image, err := m.imageStream.GetImageOfImageStream(ctx, dgst)
 	if err != nil {
+		switch err.Code {
+		case imagestream.ErrImageStreamNotFoundCode, imagestream.ErrImageStreamImageNotFoundCode:
```

Comment:
I would say ErrImageStreamImageNotFoundCode is an exceptional case, we shouldn't have dead references from image streams. So it is worth to be logged.

Reply:
Fixed